pandas/core/frame.py

```diff
@@ -1951,35 +1951,52 @@ def to_stata(
         """
         if version not in (114, 117, 118, 119, None):
             raise ValueError("Only formats 114, 117, 118 and 119 are supported.")
+        if version == 114 and convert_strl is not None:
+            raise ValueError("strl is not supported in format 114")
+
+        # TODO: There must be a better way?
```

Comment:
I think a `# type: ignore` would be reasonable here. just add the mypy error (`error: Name 'statawriter' already defined (possibly by an import)`) as a comment.

Reply:
I think an alternate since they all inherit from statawriter would be to declare something like `writer: StataWriter` before the if...else blocks and assign to that within blocks

No strong preference either way. The current method is wordier but on the plus side highlights the differences in arguments between each, which isn't bad either

Reply:
I've ignored for now and reverted this.  I can always go back to the wordier version.

Reply:
I think typing writer as StataWriter doesn't work since the different flavors accept different arguments, which triggers a warning.  I should fix that.

Reply:
...but fixing it would change the API and so I better not.